### PR TITLE
Update for latest version of widget-core

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [ ] There is a related issue
* [ ] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Update for the latest widget-core changes.

Requires release of `dojo/widget-core` and `dojo/test-extras`

### CODE REVIEW

File: src/combobox/ComboBox.ts [hunk 1]
```diff
@@ -304,10 +305,20 @@ export default class ComboBox extends ComboBoxBase<ComboBoxProperties> {
 		} = this.properties;
 
 		if (
-			!this.registry ||
+			!this._registry ||
 			includes(evt.changedPropertyKeys, 'customResultItem') ||
 			includes(evt.changedPropertyKeys, 'customResultMenu')) {
-			this.registry = this._createRegistry(customResultItem, customResultMenu);
+
+			if ( !this._registry || includes(evt.changedPropertyKeys, 'customOption')) {
```
Comment (bitpshr) [inline src/combobox/ComboBox.ts]: What is `customOption`? Do we need to check for `this._registry` twice? A little confused by these lines. Maybe they're left over from Select refactoring?

File: src/label/Label.ts [hunk 2]
```diff
@@ -45,12 +44,6 @@ export const LabelBase = ThemeableMixin(WidgetBase);
 
 @theme(baseCss)
 export default class Label extends LabelBase<LabelProperties>  {
-	@onPropertiesChanged()
-	protected onPropertiesChanged(event: PropertiesChangeEvent<this, LabelProperties>) {
```
Comment (bitpshr) [inline src/label/Label.ts]: Do we no longer need to pass a registry down to Label? I think @smhigley had some issues when wrapping a Select that uses a custom registry.

File: src/select/Select.ts [hunk 3]
```diff
@@ -226,8 +227,15 @@ export default class Select extends SelectBase<SelectProperties> {
 		} = this.properties;
 
 		// update custom option registry
-		if ( !this.registry || includes(evt.changedPropertyKeys, 'customOption')) {
-			this.registry = this._createRegistry(customOption);
+		if ( !this._registry || includes(evt.changedPropertyKeys, 'customOption')) {
+			const registry = this._createRegistry(customOption);
+			if (this._registry) {
+				this.registries.replace(this._registry, registry);
```
Comment (bitpshr) [inline src/select/Select.ts]: What's the purpose of `registries` now? I couldn't find anything in the README of widget-core yet.

File: src/combobox/ComboBox.ts (hunk 1 above)
Comment (agubler) [inline src/combobox/ComboBox.ts]: Urgh that's a fail on my part.

File: src/label/Label.ts (hunk 2 above)
Comment (agubler) [inline src/label/Label.ts]: Ah that's a reasonable point, it should use the registry mixin if that's the case (there were zero tests covering this though) or at least none failed...

File: src/select/Select.ts (hunk 3 above)
Comment (agubler) [inline src/select/Select.ts]: There is an issue open to add more about `registries` into the readme. Basically `this.registries` is access to the registry handler that manages access to registries for a widget.

File: src/label/Label.ts (hunk 2 above)
Comment (agubler) [inline src/label/Label.ts]: I have added the registry mixin to the label widget.

Comment (bitpshr) [inline src/label/Label.ts]: Ah, ok. I thought the registry mixin mandated that a `registry` is always passed.

Comment (agubler) [inline src/label/Label.ts]: It **is** actually mandatory in the registry mixin properties; I think that we will need to make this optional for the `label` and handle no registry in the registry mixin (we shouldn't copy `onPropertiesChange` code for registries around the place).

Comment (bitpshr) [inline src/label/Label.ts]: Ok cool. It worries me slightly that the Label will throw unless I pass a registry after landing this PR, so hopefully we can make the change to the registry mixin soon after.

Comment (agubler) [inline src/label/Label.ts]: Agreed; we'll sort that out beforehand.

Comment (agubler) [inline src/label/Label.ts]: In fact @bitpshr this should just work as it is now; I have made the registry optional and the registry mixin code should never be invoked if no registry property is passed.

**Pull request merged**